include/seqan3/alphabet/composition/cartesian_composition.hpp

```diff
@@ -167,37 +183,109 @@ class cartesian_composition : public pod_tuple<first_component_type, component_t
+            parent->assign_rank(
+                parent->to_rank()
+                - parent->template to_component_rank<index>() * cartesian_composition::cummulative_alph_sizes[index]
+                + to_rank() * cartesian_composition::cummulative_alph_sizes[index]);
+        }
+
+    public:
+        //Import from base type:
+        using base_t::to_rank;
+        using base_t::value_size;
+        using base_t::operator=;
+
+        /*!\name Associated types
+         * \{
+         */
+        using typename base_t::rank_type;
+        using typename base_t::char_type;
+        using typename base_t::phred_type;
+        //!\}
+
+        /*!\name Constructors, destructor and assignment
+         * \{
+         */
+        constexpr component_proxy() : base_t{}, parent{} {}
+        constexpr component_proxy(component_proxy const &) = default;
+        constexpr component_proxy(component_proxy &&) = default;
+        constexpr component_proxy & operator=(component_proxy const &) = default;
+        constexpr component_proxy & operator=(component_proxy &&) = default;
+        ~component_proxy() = default;
+
+        constexpr component_proxy(alphabet_type const l, cartesian_composition & p) :
+            base_t{l}, parent{&p}
+        {}
+
+        // Does not inherit the base's constructor for alphabet_type so as not to cause ambiguity
+        //!\}
+    };
 
     /*!\name Constructors, destructor and assignment
      * \{
      */
-    cartesian_composition() = default;
+    constexpr cartesian_composition() : base_t{} {}
     constexpr cartesian_composition(cartesian_composition const &) = default;
     constexpr cartesian_composition(cartesian_composition &&) = default;
     constexpr cartesian_composition & operator=(cartesian_composition const &) = default;
     constexpr cartesian_composition & operator=(cartesian_composition &&) = default;
     ~cartesian_composition() = default;
+
+    using base_t::base_t;
     //!\}
 
     //!\brief befriend the derived type so that it can instantiate
     //!\sa https://isocpp.org/blog/2017/04/quick-q-prevent-user-from-derive-from-incorrect-crtp-base
     friend derived_type;
 
 public:
-    //!\brief The type of value_size and `alphabet_size_v<cartesian_composition<...>>`
-    using rank_type = detail::min_viable_uint_t<(alphabet_size_v<first_component_type> * ... *
-                                                     alphabet_size_v<component_types>)>;
+    // Import from base:
+    using base_t::value_size;
+    using typename base_t::rank_type;
+    using base_t::to_rank;
+    using base_t::assign_rank;
 
-    //!\brief The product of the sizes of the individual alphabets.
-    static constexpr rank_type value_size{(alphabet_size_v<first_component_type> * ... *
-                                               alphabet_size_v<component_types>)};
+    //!\brief Export this type's components in a visible manner.
+    //!\private
+    using seqan3_tuple_element_types = component_list;
```

Comment:
It is a tuple, just the internal representation is different. The thing with `seqan3_` is a hack because we change the standard library customisation point for tuples based on this and I don't want that picking up on some other program's possible member type "tuple_element_types".
It's not pretty, but I don't have a better solution right now, maybe @rrahn knows something in his final review.